packages/wallet-sdk/src/util/checkCrossOriginOpenerPolicy.ts

```diff
@@ -0,0 +1,62 @@
+const COOP_ERROR_MESSAGE = `Coinbase Wallet SDK requires the Cross-Origin-Opener-Policy header to not be set to 'same-origin'. This is to ensure that the SDK can communicate with the Coinbase Smart Wallet app.
+
+Please see https://www.smartwallet.dev/guides/tips/popup-tips#cross-origin-opener-policy for more information.`;
+
+/**
+ * Creates a checker for the Cross-Origin-Opener-Policy (COOP).
+ *
+ * @returns An object with methods to get and check the Cross-Origin-Opener-Policy.
+ *
+ * @method getCrossOriginOpenerPolicy
+ * Retrieves current Cross-Origin-Opener-Policy.
+ * @throws Will throw an error if the policy has not been checked yet.
+ *
+ * @method checkCrossOriginOpenerPolicy
+ * Checks the Cross-Origin-Opener-Policy of the current environment.
+ * If in a non-browser environment, sets the policy to 'non-browser-env'.
+ * If in a browser environment, fetches the policy from the current origin.
+ * Logs an error if the policy is 'same-origin'.
+ */
+const createCoopChecker = () => {
+  let crossOriginOpenerPolicy: string | undefined;
+
+  return {
+    getCrossOriginOpenerPolicy: () => {
+      if (crossOriginOpenerPolicy === undefined) {
+        return 'undefined';
```

Comment:
why return a string value here? 

```suggestion
 if (crossOriginOpenerPolicy === undefined) {
        return 'undefined';
}
```

Reply:
heyy correct me if im wrong, `null` is a valid value for COOP? tested in playground, we are getting null, think here i would like to distinguish null and undefined (which is more likely an error state where checkCOOP is not called/finished)